---
title: "Lesson 7b Coding Exercises"
author: "sfhart & Katherine Brower"
date: "08/15/2023"
output: html_document
---


## Data: The Cancer Genome Atlas (TCGA) bladder cancer sequencing and patient
##       characteristics


## Graphing using ggplot2

## Instructions:
## Work with a partner or two!:
## - We've looked at a number of variables in our RB1_clinical dataset. Choose
##   two variables (columns) you think might have an interesting relationship.
## - Note: 
##   - it may be wise to avoid relating 2 categorical variables (may not be able to plot it!)
## - Form a hypothesis about these variables and discuss with your group which plot type would best help you evaluate it.
## - Generate the plot using the techniques we learning in ggplot.
## - Change the color/fill of at least one part of the plot and use
##    coord_cartesian to "zoom" in/out as necessary.
## - Add informative x- and y-axis labels.
## - Use ggsave() to save to plot to an appropriate name in the plots directory (folder)


```{r}
# install.packages("tidyverse")
library(tidyverse) ## if this isn't working, be sure to run the above (uncomment first)
```

# Run the following code from lesson 6 to make RB1_clinical
```{r}
clinical <- read_csv("data/clinical.csv")
RB1_clinical <- clinical %>%
  select(tissue_or_organ_of_origin:total.number.Indels, contains("RB1")) %>%
  filter(sample_type == "Primary Tumor") %>%
  mutate(across(c(focal.deletion.in.RB1,mutation.in.RB1), #find cols that contain "RB1"
                ~replace_na(., "NaN"))) %>%  #replace the NAs with strings
  mutate(any.RB1.mut = focal.deletion.in.RB1 == "Del" |
           focal.deletion.in.RB1 == "Loss" |
           mutation.in.RB1 == "yes"
  ) %>%
  mutate(RB1_expr = as.numeric(RB1_expr)) #make sure expression is numeric just in case
```


```{r}
# [Add code here]
```



