---
title: "Plots 2.0 Challenge Problems KEY"
author: "sfhart & Katherine Brower & Hanako Osuga"
date: "2.21.24"
output: html_document
---


## Data: The Cancer Genome Atlas (TCGA) bladder cancer sequencing and patient
##       characteristics


## Graphing using ggplot2

## Instructions:
## Work with a partner or two!:
## - We've looked at a number of variables in our RB1_clinical dataset. Choose
##   two variables (columns) you think might have an interesting relationship.
## - Note: 
##   - it may be wise to avoid relating 2 categorical variables (may not be able to plot it!)
## - Form a hypothesis about these variables and discuss with your group which plot type would best help you evaluate it.
## - Generate the plot using the techniques we learning in ggplot.
## - Change the color/fill of at least one part of the plot and use
##    coord_cartesian to "zoom" in/out as necessary.
## - Add informative x- and y-axis labels.
## - Use ggsave() to save to plot to an appropriate name in the plots directory (folder)


```{r}
# install.packages("tidyverse")
library(tidyverse) ## if this isn't working, be sure to run the above (uncomment first)
```

# Run the following code we wrote previously to make RB1_clinical
```{r}
clinical <- read_csv("data/clinical.csv")
RB1_clinical <- clinical %>%
  select(tissue_or_organ_of_origin:total.number.Indels, contains("RB1")) %>%
  filter(sample_type == "Primary Tumor") %>%
  mutate(across(c(focal.deletion.in.RB1,mutation.in.RB1), #find cols that contain "RB1"
                ~replace_na(., "NaN"))) %>%  #replace the NAs with strings
  mutate(any.RB1.mut = focal.deletion.in.RB1 == "Del" |
           focal.deletion.in.RB1 == "Loss" |
           mutation.in.RB1 == "yes"
  ) %>%
  mutate(RB1_expr = as.numeric(RB1_expr)) #make sure expression is numeric just in case
```

# Make a plot of the two variables you find interesting
```{r}
#example with Tobacco.smoking.hustory and RB1_expr
A_boxplot <- ggplot(data = RB1_clinical,
       aes(x = Tobacco.smoking.history,
           y = RB1_expr)) + 
  geom_boxplot(outlier.shape = NA) +
  coord_cartesian(ylim = c(0, 25)) +
  geom_jitter(color = "blue") +
  theme_grey()+
    theme(axis.text.y = element_text(color = "black", size = 7),
        axis.text.x = element_text(size = 7, color = "black", angle = 45, hjust = 1),
        axis.title.y = element_text(color = "black", size = 7),
        axis.title.x = element_text(color = "black", size = 7),
        legend.position = "none") +
  labs(title = "Tobacco smoking history vs. RB1_expr",
       x = "Tobacco smoking history",
       y = "RB1 Expression")
A_boxplot

ggsave(filename = "plots/smoking_history_vs_RB1_Expr.pdf",
       plot = A_boxplot,
       width = 6, 
       height = 12) 
```


# PART 2

All exercises use the clinical object we created in the readings:

```{r}
# install.packages("tidyverse")
library(tidyverse) ## if this isn't working, be sure to run the above (uncomment first)

# Load the data
clinical <- read.csv("data/clinical.csv")
```

This is code to make a boxplot:
```{r}
any_RB1_mut <- ggplot(data = RB1_clinical, aes(x = any.RB1.mut, y = RB1_expr)) +
  geom_boxplot() +
  coord_cartesian(ylim = c(0, 20)) +
  geom_jitter(aes(color = any.RB1.mut), alpha = 0.5) +
  labs(title = "RB1 mutations and RB1 expression",
       x = "RB1 Mutation Presence",
       y = "RB1 expression")

any_RB1_mut


```

Code jitter before the boxplot. What happens?
```{r}
any_RB1_mut <- ggplot(data = RB1_clinical, aes(x = any.RB1.mut, y = RB1_expr)) +
  geom_jitter(aes(color = any.RB1.mut), alpha = 0.5) +
  geom_boxplot() +
  coord_cartesian(ylim = c(0, 20)) +
  labs(title = "RB1 mutations and RB1 expression",
       x = "RB1 Mutation Presence",
       y = "RB1 expression")

any_RB1_mut

```

Layer the code block using theme(), modifying the axes labels and the legend.
```{r}

any_RB1_mut <- ggplot(data = RB1_clinical, aes(x=any.RB1.mut, y=RB1_expr)) +
  geom_boxplot()+
  coord_cartesian(ylim=c(0,20)) +
  geom_jitter(aes(color = any.RB1.mut), alpha = 0.5) +
  theme_bw() +
  theme(axis.text.y = element_text(color = "black", size = 7),
        axis.text.x = element_text(size = 7, color = "black", angle = 45, hjust = 1),
        axis.title.y = element_text(color = "black", size = 7),
        axis.title.x = element_text(color = "black", size = 7),
        legend.position = "none") +
  labs(title = "RB1 mutations and RB1 Expression",
       x = "RB1 Mutations",
       y = "RB1 Expression")


any_RB1_mut


```
    
Now flip the order so that you put the labels before theme_bw(), what happens?    
Given your observations after doing the above, does the order of layers matter?
```{r}

any_RB1_mut <- ggplot(data = RB1_clinical, aes(x=any.RB1.mut, y=RB1_expr)) +
  geom_boxplot()+
  coord_cartesian(ylim=c(0,20)) +
  geom_jitter(aes(color = any.RB1.mut), alpha = 0.5) +
  labs(title = "RB1 mutations and RB1 Expression",
       x = "RB1 Mutations",
       y = "RB1 Expression")+
  theme_bw() +
  theme(axis.text.y = element_text(color = "black", size = 7),
        axis.text.x = element_text(size = 7, color = "black", angle = 45, hjust = 1),
        axis.title.y = element_text(color = "black", size = 7),
        axis.title.x = element_text(color = "black", size = 7),
        legend.position = "none") 


any_RB1_mut


```
    

In the box plot above, use theme_gray() (or theme_grey()) instead of theme_bw(). 

What do you observe when theme_gray() is present vs when it is absent in the code block? What can you conclude about default theme settings in ggplot2?
```{r}

any_RB1_mut <- ggplot(data = RB1_clinical, aes(x=any.RB1.mut, y=RB1_expr)) +
  geom_boxplot()+
  coord_cartesian(ylim=c(0,20)) +
  geom_jitter(aes(color = any.RB1.mut), alpha = 0.5) +
  theme_grey() +
  theme(axis.text.y = element_text(color = "black", size = 7),
        axis.text.x = element_text(size = 7, color = "black", angle = 45, hjust = 1),
        axis.title.y = element_text(color = "black", size = 7),
        axis.title.x = element_text(color = "black", size = 7),
        legend.position = "none") +
  labs(title = "RB1 mutations and RB1 Expression",
       x = "RB1 Mutations",
       y = "RB1 Expression")

any_RB1_mut

#the default if theme_grey!

```

In the Scatterplots and Boxplots lesson, we created a scatterplot relating smoking to the number of ERCC2 signature mutations. Use what we learned in the lesson to create a more presentable plot.
    Set the colors of the points to a column with continuous (numerical) data of your choice. What happens? Is there a trend?
   
```{r}

#example for cigarettes_per_day 
d_plot <- clinical %>% 
  select(barcode,ajcc_pathologic_stage, cigarettes_per_day, count.ERCC2.signature.mutations) %>% 
  filter(!is.na(count.ERCC2.signature.mutations)) 
 

scatter_1<- ggplot(data = d_plot,
                   aes(x = cigarettes_per_day,
                       y = count.ERCC2.signature.mutations,
                       fill = cigarettes_per_day)) + 
  geom_point(color = "pink") +
  coord_cartesian(ylim = c(0, 400), xlim=c(0,8)) +
  theme_grey() +
  theme(axis.text.y = element_text(color = "black", size = 7),
        axis.text.x = element_text(size = 7, color = "black", hjust = 1),
        axis.title.y = element_text(color = "black", size = 7),
        axis.title.x = element_text(color = "black", size = 7),
        legend.position = "none") +
  labs(title = "ERCC2 mutations vs cigarettes per day",
       x = "Cigarettes per day",
       y = "# of ERCC2 mutations")

scatter_1

```

Set the colors of the points to be by cancer stage category (relevant column is called ajcc_pathologic_stage). Is there a trend? 
```{r}
stage_ERCC2 <- ggplot(data = d_plot,
                      aes(x = cigarettes_per_day,
                          y = count.ERCC2.signature.mutations,
                          fill = ajcc_pathologic_stage)) + 
  geom_point(aes(color = ajcc_pathologic_stage)) +
  coord_cartesian(ylim = c(0, 125), xlim = c(0,7)) +
  theme_bw() +
  theme(axis.text.y = element_text(color = "black", size = 7),
        axis.text.x = element_text(size = 7, color = "black", angle = 45, hjust = 1),
        axis.title.y = element_text(color = "black", size = 7),
        axis.title.x = element_text(color = "black", size = 7))+
  labs(title = "ERCC2 mutations vs Cancer Stage Category",
       x = "Cigarettes per day",
       y = "# of ERCC2 mutations")
stage_ERCC2
```


Create a plot of your choice using the data in clinical. Use what you learned today to add appropriate customization.
```{r}
#an example with total.number.Indels and year of birth 

my_plot <- ggplot( data = clinical, 
                   aes(x= year_of_birth, 
                       y= total.number.Indels)) +
  geom_point(color =  "#d8c0e4" ) +
  coord_cartesian(ylim = c(0,30)) +
  theme_grey()+
    theme(axis.text.y = element_text(color = "black", size = 10),
        axis.text.x = element_text(size = 8, color = "black"),
        axis.title.y = element_text(color = "black", size = 8),
        axis.title.x = element_text(color = "black", size = 8),
        legend.position = "none") +
    labs(title = "Total Number of Indels by Year of Birth",
         x = "Year of Birth",
         y = "Total Number of Indels")
           
my_plot


#same plot colored by cancer stage

my_plot <- ggplot( data = clinical, 
                   aes(x= year_of_birth, 
                       y= total.number.Indels)) +
  geom_point(aes(color = ajcc_pathologic_stage) ) +
  coord_cartesian(ylim = c(0,30)) +
  theme_grey()+
    theme(axis.text.y = element_text(color = "black", size = 10),
        axis.text.x = element_text(size = 8, color = "black"),
        axis.title.y = element_text(color = "black", size = 8),
        axis.title.x = element_text(color = "black", size = 8) )+
    labs(title = "Total Number of Indels by Year of Birth",
         x = "Year of Birth",
         y = "Total Number of Indels")
           
my_plot

```
Save this plot to turn in! Make the name of the file your name.
```{r}
ggsave(filename = "plots/Your_Name_Here.pdf",
       plot = my_plot,
       width = 6, 
       height =12) 

```

