---
title: "Data Manipulation"
author: "Sam Hart & Katherine Brower"
date: "2023-08-10"
output: html_document
---

__Under each heading below, type your code and run it. Feel free to add notes and comments to help you learn.__ 


### load library/package

```{r}
library(tidyverse)
```


### import data
```{r}
clinical <- read_csv("clinical.csv")
```

### show categories in gender
```{r}
unique(clinical$gender) # remember, the $ extracts the column as a vector
```


### convert the column, age_at_diagnosis from days to years using pipes, as a new column called clinical_years 
```{r}

```

### determine the total small mutations (nucleotide variants and insertions/deletions). name this new column total_small_mutations
### insertions and deletions are combined and called indels in this dataset
```{r}


```

### combine the columns, ATM_expr and KMT2D_expr together as a new column called ATM_KMT2D_total_expr
```{r}

```
### summarize average birth year by gender
```{r}
avg_YOB_by_gender <- 
```

### summarize average age at diagnosis (in days) by gender
```{r}
avg_age_days_by_gender <-
```

### count number of individuals by gender and arrange in ascending order
```{r}
gender_count <- 

```

### use the %in% operator to extract data specifying primary diagnoses to keep: Transitional cell carcinoma and Papillary transitional cell carcinoma
```{r}

```
