---
title: "Scatterplots and Boxplots Exercises KEY"
author: "sfhart & Katherine Brower"
date: "08/14/2023"
output: html_document
---

## Data: The Cancer Genome Atlas (TCGA) bladder cancer sequencing and patient
##       characteristics
```{r}
# install.packages("tidyverse")
library(tidyverse) ## if this isn't working, be sure to run the above (uncomment first)
```


## Graphing using ggplot2

#  Instructions:
# Let's revisit the relationship between RB1 expression and the types of
# mutations in RB1 that we've seen in our bladder cancer samples!


# to start, let's remake the subset RB1_clinical that we made in the piping exercise
# RB1_clinical should have data from clinical that selects for the columns tissue_or_organ_of_origin through
# total.number.Indels that contain "RB1" (we want to find columns that contain RB1)
# it should also filter for any rows that has Primary Tumor as the sample type

```{r}
#clinical <- read_csv("data/clinical.csv")

RB1_clinical <- clinical %>%
  select(tissue_or_organ_of_origin:total.number.Indels, contains("RB1")) %>%
  filter(sample_type == "Primary Tumor") 

```

# let's add on to the code above to do the following:
# 1) replace the NAs in focal.deletion.in.RBA and mutation.in.RB1 with "NaN"
# 2) create a new column called "any.RB1.mut"; set it equal to having "Del" or "Loss" in focal.deletion.in.RB1 OR "yes" in mutations.in.RB1 
# 3) double check RB1_expr is a numeric, just in case

```{r}
#clinical <- read_csv("data/clinical.csv")

RB1_clinical <- clinical %>%
  select(tissue_or_organ_of_origin:total.number.Indels, contains("RB1")) %>%
  filter(sample_type == "Primary Tumor") %>%
  
  mutate(across(c(focal.deletion.in.RB1,mutation.in.RB1), #find cols that contain "RB1"
                ~replace_na(., "NaN"))) %>%  #replace the NAs with strings
  mutate(any.RB1.mut = focal.deletion.in.RB1 == "Del" |
           focal.deletion.in.RB1 == "Loss" |
           mutation.in.RB1 == "yes"
  ) %>%
  mutate(RB1_expr = as.numeric(RB1_expr)) #make sure expression is numeric just in case
```


# Now that we have the data we need, lets make our plot
#  Make a boxplot comparing RB1 expression between cancers that have any
#  RB1 mutation and those that do not. Use the RB1_clinical dataset you created previously.
#  Hint: Which column holds logical values for our RB1 mutations of interest? (e.g. use names() or View())
# Call this any_RB1_mut

```{r}
#RB1_clinical
#names(RB1_clinical)
any_RB1_mut <- ggplot(data = RB1_clinical, aes(x=any.RB1.mut, y=RB1_expr)) +
  geom_boxplot()

any_RB1_mut
```

# Reflection Questions
##    We were aware of the difference in mean/medians, but what can we now
##      say about the range, outliers, etc.? What new questions/hypothesis can
##      we generate from these data?


##  - Then use coord_cartesian() to "zoom" to between 0 and 20 FPKM (the units
##    for this analysis)
```{r}
any_RB1_mut <- ggplot(data = RB1_clinical, aes(x=any.RB1.mut, y=RB1_expr)) +
  geom_boxplot()+
  coord_cartesian(ylim=c(0,20))

any_RB1_mut
```

# don't forget to save your plot to your plots directory!
```{r}
ggsave( filename = "plots/anyRB1mut.PDF",
        plot = any_RB1_mut,
        width = 4, height = 3)
```


#  Reflection Questions
#  How does this new view alter what questions we might generate?


# now lets use RB1_clinical to make a scatterplor
# make a scatterplot of RB1_expr and age_at_diagnosis
```{r}
RB1_age_plot <- ggplot(data = RB1_clinical,
                      mapping = aes(x= RB1_expr, y = age_at_diagnosis))+
  geom_point()

RB1_age_plot 
```


# lets remove the outlier by zooming into 0 to 30
```{r}
RB1_age_plot <- ggplot(data = RB1_clinical,
                      mapping = aes(x= RB1_expr, y = age_at_diagnosis))+
  geom_point() +
  coord_cartesian(xlim = c(0,30))

RB1_age_plot 
```

# we still have that second outlier, change the coord_cartesian again to remove it
```{r}
RB1_age_plot <- ggplot(data = RB1_clinical,
                      mapping = aes(x= RB1_expr, y = age_at_diagnosis))+
  geom_point() +
  coord_cartesian(xlim = c(0,25))

RB1_age_plot 
```


# save the plot to your plots directory!
```{r}
ggsave( filename = "plots/RB1expr-age.PDF",
        plot = RB1_age_plot,
        width = 4, height = 3)
```

