---
title: "Scatterplots and Boxplots Challenge Problems"
author: "sfhart & Katherine Brower & Hanako Osuga"
date: "2/21/2024"
output: html_document
---

Don't forget to load these if you haven't already!
```{r}
# install.packages("tidyverse")
library(tidyverse) ## if this isn't working, be sure to run the above (uncomment first)

# Load the data
clinical <- read.csv("data/clinical.csv")
```


Create a visualization (plot) showing the effects of mutations on RNA expression level (expression levels are given in ELF3, relevant columns end in “expr”). What do you see? Hypothesize what this tells you about the effect of the mutation.
```{r}

KMT2D_plot <- ggplot( data = clinical, 
                     aes( x= mutation.in.KMT2D,
                          y = KMT2D_expr)) +
  geom_boxplot()+ 
  geom_jitter(color = 	"#d0457a") +
  coord_cartesian(ylim=c(0, 20)) + 
  theme_grey()+
    theme(axis.text.y = element_text(color = "black", size = 7),
          axis.text.x = element_text(size = 7, color = "black", angle = 45, hjust = 1),
          axis.title.y = element_text(color = "black", size = 7),
          axis.title.x = element_text(color = "black", size = 7),
          legend.position = "none") +
    labs(title = "Expression of KMT2D across samples with and without mutations",
         x = "Presence of Mutation",
         y = "KMT2D Expression")

KMT2D_plot



```

Does smoking have an effect on the number of mutations? Create visualizations and investigate if smoking is correlated to:
    The number of single nucleotide variants (also known as point mutations, mutations in just one base) 
    The number of indels (mutations that are insertion or deletions)
    Total number of mutations: create a column summing up the number of point mutations and indels.
```{r}

# total SNPs
SNP_plot <- ggplot( data= clinical,
                    aes( x = Tobacco.smoking.history,
                    y = total.number.Single.Nucleotide.Variants)) +
  geom_boxplot()+
  geom_jitter( color = "#235347") +
  coord_cartesian(ylim=c(0, 2000)) +
  theme_grey() +
    theme(axis.text.y = element_text(color = "black", size = 7),
          axis.text.x = element_text(size = 7, color = "black", angle = 45, hjust = 1),
          axis.title.y = element_text(color = "black", size = 7),
          axis.title.x = element_text(color = "black", size = 7),
          legend.position = "none") +
    labs(title = "SNPs and Smoking History",
         x = "Smoking History",
         y = "Total number of SNPs")
SNP_plot


```
```{r}

# total indels 
indel_plot <- ggplot( data = clinical,
                      aes( x = Tobacco.smoking.history,
                           y = total.number.Indels)) +
  geom_boxplot() +
  geom_jitter( color = "#1c4966") +
  coord_cartesian(ylim=c(0, 40)) +
  theme_grey() +
    theme(axis.text.y = element_text(color = "black", size = 7),
          axis.text.x = element_text(size = 7, color = "black", angle = 45, hjust = 1),
          axis.title.y = element_text(color = "black", size = 7),
          axis.title.x = element_text(color = "black", size = 7),
          legend.position = "none") +
    labs(title = "Indels and Smoking History",
         x = "Smoking History",
         y = "Total number of Indels")
indel_plot


```




```{r}
# total number of mutations

#clean the data and make a new column total.Mutations
mutations_clinical <- clinical %>%
  select(Tobacco.smoking.history, total.number.Indels, total.number.Single.Nucleotide.Variants) %>%
  mutate(total_Indels_noNA = replace_na(total.number.Indels, 0)) %>%  #replace the NAs with 0
  mutate(total_SNPs_noNA = replace_na(total.number.Single.Nucleotide.Variants, 0)) %>%  #replace the NAs with 0
  mutate(total.Mutations = total_Indels_noNA + total_SNPs_noNA) #add the number of Indels and SNPs together, store value in total.Mutations


mutations_plot <- ggplot(data= mutations_clinical,
                         aes(x = Tobacco.smoking.history,
                              y = total.Mutations)) +
  geom_boxplot() +
  geom_jitter(color = "#ffb900") +
  coord_cartesian(ylim=c(0, 2000)) +
  theme_grey() +
    theme(axis.text.y = element_text(color = "black", size = 7),
          axis.text.x = element_text(size = 7, color = "black", angle = 45, hjust = 1),
          axis.title.y = element_text(color = "black", size = 7),
          axis.title.x = element_text(color = "black", size = 7),
          legend.position = "none") +
    labs(title = "All mutations (indels and SNPs) and Smoking History",
         x = "Smoking History",
         y = "Total number of mutations")
  
mutations_plot

```


Some mutations are named “focal deletion” and “focal amplification” in clinical. Choose a gene of interest (e.g. focal.amplification.in.E2F3) and relate its mutational status to RNA levels (e.g. E2F3_expr). What can you hypothesize about the effect of these categories of mutations on RNA levels (transcription)?
    Note that for focal amplification:
        "." means no amplification   
        "Gain" and "Amp" mean focal amplification is present 
        NA and "NaN" mean missing data
```{r}

#example for focal.amplification.in/E2F3

#clean the data 
E2F3_clinical <- clinical %>%
  select(focal.amplification.in.E2F3, E2F3_expr) %>%
  mutate(E2F3_amp = replace(focal.amplification.in.E2F3, focal.amplification.in.E2F3== ".", "No Amp")) %>% # replace . for No Amp
  mutate(E2F3_amp_noNA = replace(E2F3_amp, focal.amplification.in.E2F3 == "NaN", NA) ) # replace NaN with NA


E2F3_plot <- ggplot(data = E2F3_clinical, 
                    aes(x= E2F3_amp_noNA,
                        y = E2F3_expr))+
  geom_boxplot()+
  geom_jitter(color = "#710000") +
  coord_cartesian(ylim = c(0,120)) +
  theme_grey()+
   theme(axis.text.y = element_text(color = "black", size = 7),
          axis.text.x = element_text(size = 7, color = "black", angle = 45, hjust = 1),
          axis.title.y = element_text(color = "black", size = 7),
          axis.title.x = element_text(color = "black", size = 7),
          legend.position = "none") +
    labs(title = "Focal amplification in E2F3 and expression",
         x = "Focal amplification",
         y = "E2F3 expression")
E2F3_plot


```

Explore the data on your own! Create your own plot using two numerical variables from the data in clinical and see if you can find interesting signals in the data. You may choose to try other plot types such as histograms using geom_histogram()

```{r}

#example of cigarettes_per_day and pack_years_smoked, colored by gender 
smoking_plot <- ggplot(data = clinical,
                      aes(x = cigarettes_per_day,
                          y = pack_years_smoked)) + 
  geom_point(aes(color = gender)) +
  coord_cartesian(ylim = c(0, 125), xlim = c(0,7)) +
  theme_bw() +
  theme(axis.text.y = element_text(color = "black", size = 7),
        axis.text.x = element_text(size = 7, color = "black", hjust = 1),
        axis.title.y = element_text(color = "black", size = 7),
        axis.title.x = element_text(color = "black", size = 7)) +
  labs(title = "Cigarettes per day vs pack years smoked",
       x = "Cigarettes per day",
       y = "Pack years smoked")
smoking_plot

#why does this plot look this way? Does it make sense with the data you plotted?

```
