---
title: "8.1_piping_Exercises.Rmd"
output: html_document
date: "2024-01-24"
---

# Data: The Cancer Genome Atlas (TCGA) bladder cancer sequencing and patient characteristics

## Import packages

```{r}
# install.packages("tidyverse")
library(tidyverse) ## if this isn't working, be sure to run the above (uncomment first)
```

## Load the data

```{r}
clinical <- read.csv("data/clinical.csv")

# this is the selection procedure you used yesterday to select a subset of the columns
RB1_clinical <- select(clinical,
                       tissue_or_organ_of_origin:total.number.Indels,
                       contains("RB1"))
```


# Part I: Piping practice

# filter RB1_clinical for just the "Primary Tumor" samples types in "sample_type"
# call this RB1_trial1 
```{r}
Rb1_trial1 <- filter(RB1_clinical, sample_type== "Primary Tumor")
```


# filter RB1_clinical for patients who have either "Del" or "Loss" in focal.deletion.in.RB1 OR "yes" in mutations.in.RB1 
# call this RB1_trial2
# hint: we can use the "|" operator for OR. 
```{r}
RB1_trial2 <- filter(RB1_clinical, focal.deletion.in.RB1 == "Del" | focal.deletion.in.RB1 == "Loss" | mutation.in.RB1 == "yes")
```


# use the pipe operator to combine the selecting/filtering steps above
# call this RB1_trial3
# you'll need a total of three expressions in your filtering step using the form: 

my_data_frame %>%
  fuction(inputs) %>%
  function(inputs)
  
```{r}
RB1_trial3 <- RB1_clinical %>%
  filter(sample_type == "Primary Tumor") %>%
  filter(
    focal.deletion.in.RB1 == "Del" |
    focal.deletion.in.RB1 == "Loss" |
    mutation.in.RB1 == "yes"
  )
```


# Part 2: Working with dataframes, mutate and pipe

Because RB1 is a tumor suppressor (when it's present, it helps prevent cells from becoming cancerous) we mostly care about whether the gene is "broken" regardless of how it happened. 

Use pipes to do the following:

# use RB1_clinical and create a new column called "any.RB1.mut"; set it equal to the statement that you used to filter earlier (the three equals expressions. This will create a logical column for whether there's any mutation in RB1 in the cancers.
```{r}
RB1_trial <- RB1_clinical %>%
  mutate(any.RB1.mut = (focal.deletion.in.RB1 == "Del") |
           (focal.deletion.in.RB1 == "Loss") |
           (mutation.in.RB1 == "yes"))

```

# create two new dataframes: one that contains TRUEs for this new col and one that contains just FALSEs. Name them something you can remember. Hint: By "create" we just mean to assign it to a variable. You'll need to use one of the functions we learned today to limit the rows to just the ones that meet the criteria (i.e. either TRUE or FALSE)
```{r}
# Cancers with an RB1_mutation
RB1_mutants <- RB1_clinical %>%
  mutate(any.RB1.mut = focal.deletion.in.RB1 == "Del" |
           focal.deletion.in.RB1 == "Loss" |
           mutation.in.RB1 == "yes"
  ) %>%
  filter(any.RB1.mut == TRUE) 

## "Wild type" (aka no mutation in RB1) cancers
RB1_wildtype <- RB1_clinical %>%
  mutate(any.RB1.mut = focal.deletion.in.RB1 == "Del" |
           focal.deletion.in.RB1 == "Loss" |
           mutation.in.RB1 == "yes"
  ) %>%
  filter(any.RB1.mut == FALSE) 
```


# extract the "RB1_expr" column as a vector and assign it to variable for each of your two new dataframes (created above). Hint: Review the vectors lesson if you don't remember how to extract a column as a vector.
```{r}
expr.RB1_mutants <- RB1_mutants$RB1_expr

expr.RB1_wildtype <- RB1_wildtype$RB1_expr
```

# does the median expression of RB1 (RB1_expr) differ between these groups?
```{r}
median(expr.RB1_mutants)
# 3.415344

median(expr.RB1_wildtype)
# 8.302018

```

# does the mean expression of RB1 (RB1_expr) differ between these groups?
```{r}
mean(expr.RB1_mutants)
# 4.224855

mean(expr.RB1_wildtype)
# 8.752361

```
 
Why might this result be, based on your knowledge of gene structure and types of cancer-relevant mutations?

# RB1_mutants are ones that lost or deleted the RB1 gene, if the gene is lost or deleted it might not be expressed as much as the wild type (when gene is present). This would explian the difference in mean and median expression of RB1. 

RB1 is a tumor suppressor, meaning that if it's there it can help prevent cancers from developing. Would you expect a higher rate of cancer in th expr.RB1_wildtype or expr.RB1_mutants?

# expr.RB1_mutants! The brakes on this car are broken or missing. 

