# 8.2 Piping Exercises

Name:

Date:

------------------------------------------------------------------------

In this exercise, we will practice piping with data from The Cancer Genome Atlas (TCGA) bladder cancer sequencing and patient characteristics.

We will start with a fresh, clean environment. The code chunk below clears your environment if you want to reset your work in interactive mode. Or use the broom button on the top right corner.

```{r}
rm(list = ls())
```

## Load the data and packages

```{r}
library(tidyverse)
```

```{r}
clinical <- read.csv("data/clinical.csv")

# This is the selection procedure you used yesterday to select a subset of the columns
RB1_clinical <- select(clinical,
                       tissue_or_organ_of_origin:total.number.Indels,
                       contains("RB1"))
```

Let's look at our data:

```{r}
head(RB1_clinical)
```

------------------------------------------------------------------------

## Part I: Piping practice

1.  Filter `RB1_clinical` for just the "Primary Tumor" sample types in the `sample_type` column.

    ```{r}

    ```

2.  Filter `RB1_clinical` for patients who have either "Del" or "Loss" in `focal.deletion.in.RB1` OR "yes" in `mutations.in.RB1`.

    *Hint: We can use the `|` operator for OR.*

    ```{r}

    ```

3.  Use the pipe operator to combine the selecting/filtering steps above. You'll need a total of three expressions using the form:

    `my_data_frame %>% function(inputs) %>% function(inputs)`

    ```{r}

    ```

------------------------------------------------------------------------

## Part II: Working with dataframes, mutate and pipe

Because RB1 is a tumor suppressor (when it's present, it helps prevent cells from becoming cancerous) we mostly care about whether the gene is "broken" regardless of how it happened.

1.  Use pipes with `RB1_clinical` to create a new column called `any.RB1.mut`. Set it equal to the logical statement you used to filter earlier (the three equals expressions). This will create a logical column for whether there's any mutation in RB1 in the cancers.

    ```{r}

    ```

2.  Create two new dataframes: one that contains `TRUE` values for this new column and one that contains just `FALSE` values. Name them something you can remember.

    *Hint: By "create" we just mean to assign it to a variable. You'll need to use `filter()` to limit the rows to just the ones that meet the criteria (either TRUE or FALSE).*

    ```{r}

    ```

3.  Extract the `RB1_expr` column as a vector and assign it to a variable for each of your two new dataframes (created above).

    *Hint: Review the vectors lesson if you don't remember how to extract a column as a vector.*

    ```{r}

    ```

4.  Does the median expression of RB1 (`RB1_expr`) differ between these groups?

    ```{r}

    ```

5.  Does the mean expression of RB1 (`RB1_expr`) differ between these groups?

    ```{r}

    ```

------------------------------------------------------------------------

## Part III: Reflection questions

1.  Why might this result be, based on your knowledge of gene structure and types of cancer-relevant mutations?

    ```         
    ```

2.  RB1 is a tumor suppressor, meaning that if it's there it can help prevent cancers from developing. Would you expect a higher rate of cancer in the RB1_wildtype or RB1_mutants?

    ```         
    ```
